### PR TITLE
[PXZ-19] User Management

### PR DESCRIPTION
**Pull request by juliabaptista**
# [PXZ-19] User Management

## Technical Tasks:
- [x] User Model -> User Entity
- [x] Role Enum
- [x] User Repository
- [x] User Resgistration DTO

## Technical Tasks - NEED TO BE IMPLEMENTED:
- [ ] User Controller
- [ ] User Service
- [ ] User Registration
- [ ] Configure Spring Security for User Authentication
- [ ] Define User Roles and Permissions
- [ ] User Login with JWT

[PXZ-19]: https://jbptst.atlassian.net/browse/PXZ-19?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by faruka-startsteps** [comment]
should the target branch be dev?

**Comment by juliabaptista** [comment]
> should the target branch be dev? 

thanks @faruka-startsteps for pointing this
done ✅

File: src/main/java/de/zalando/ECommerceShopApplication.java [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
 import io.swagger.v3.oas.annotations.OpenAPIDefinition;
 import io.swagger.v3.oas.annotations.info.Info;
+import jakarta.persistence.*;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 import org.springframework.web.bind.annotation.RestController;
```
Comment (moseseth) [inline src/main/java/de/zalando/ECommerceShopApplication.java:12]: Why do you need EntityScan and EnableJpaRepsitory

File: src/main/java/de/zalando/dto/UserRegistrationRequest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  @NotEmpty(message = "Required field. Cannot be empty!")
+  private String lastName;
+
+  @Email(message = "Please, provide a valid email address!")
+  @NotNull(message = "Required field. Cannot be null!")
+  @NotEmpty(message = "Required field. Cannot be empty!")
+  private String email;
+
+  @NotNull(message = "Required field. Cannot be null!")
+  @NotEmpty(message = "Required field. Cannot be empty!")
```
Comment (moseseth) [inline src/main/java/de/zalando/dto/UserRegistrationRequest.java:29]: @NotEmtpy covers both at not null and blank cases..So, you can remove @NotNull

**Review by moseseth — CHANGES_REQUESTED** [review]

File: src/main/java/de/zalando/ECommerceShopApplication.java (hunk 1 above)
Comment (juliabaptista) [reply src/main/java/de/zalando/ECommerceShopApplication.java:12]: Already deleted @moseseth 😉 It was my mistake to keep all there.
Screenshot 2024-01-22 at 16 48 19

File: src/main/java/de/zalando/dto/UserRegistrationRequest.java (hunk 2 above)
Comment (juliabaptista) [reply src/main/java/de/zalando/dto/UserRegistrationRequest.java:25]: @moseseth thanks! 🙂 
Didn't know that! Fix it ✅

**Review by moseseth — APPROVED** [review]

**Pull request merged**